### PR TITLE
Implement `$project` pipeline aggregation stage

### LINKED ISSUE #2382: Implement `$project` pipeline stage
### What should be done?

https://www.mongodb.com/docs/manual/reference/operator/aggregation/project/

We have to implement `$project` pipeline stage first in order to add support for #2317.

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

Closes #2382.

## Readiness checklist

* [ ] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/common/aggregations/stages/project.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+type project struct {
+	projection *types.Document
+	inclusion  bool
+}
+
+// newProject validates projection document and creates a new $project stage.
+func newProject(stage *types.Document) (Stage, error) {
+	fields, err := common.GetRequiredParam[*types.Document](stage, "$project")
+	if err != nil {
+		return nil, err
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/stages/project.go:41]: Could you add a test for `$project` not being document? Something like below would give us error message here.

```
pipeline: bson.A{bson.D{{"$project", "invalid"}}}
```

File: internal/handlers/common/aggregations/stages/project.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	validated, inclusion, err := common.ValidateProjection(fields)
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/stages/project.go:44]: The error from `NewCommandErrorMsgWithArgument` has argument of `projection`, we want to have `$project (stage)` if we can.

**Review by chilagrow — COMMENTED** [review]
Looks great, just tiny comments

**Review by noisersup — DISMISSED** [review]
Apart from comment above looks good to me!

File: internal/handlers/common/aggregations/stages/project.go (hunk 1 above)
Comment (w84thesun) [reply internal/handlers/common/aggregations/stages/project.go]: Nice catch, thanks! Added this one.

File: internal/handlers/common/aggregations/stages/project.go (hunk 2 above)
Comment (w84thesun) [reply internal/handlers/common/aggregations/stages/project.go:53]: I unwrapped the error and returned the correct one below.

**Review by noisersup — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**